### PR TITLE
Feature/add angular

### PR DESCRIPTION
**Pull request by ShugKnight24**
- Add Angular & connect to Nest server
- Add Login & Register forms
- Add custom validators for password & password matching

### CODE REVIEW

File: blog/.vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (csalajan) [inline blog/.vscode/extensions.json:1]: I would recommend adding this entire .vscode directory to your .gitignore.

File: blog/src/app/components/register/register.component.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  FormGroup,
+  ValidationErrors,
+  Validators
+} from '@angular/forms';
+import { Router } from '@angular/router';
+import { map } from 'rxjs/internal/operators/map';
+import { AuthenticationService } from 'src/app/services/authentication.service';
+
+class CustomValidators {
+  static passwordContainsNumber(
```
Comment (csalajan) [inline blog/src/app/components/register/register.component.ts:14]: I would recommend abstracting some of this logic to be less "Password" specific, and create it as a reusable utility

File: blog/.vscode/extensions.json (hunk 1 above)
Comment (ShugKnight24) [reply blog/.vscode/extensions.json]: resolved by [fa6aa13](https://github.com/ShugKnight24/angular_nest_blog/pull/8/commits/fa6aa13c9516ce658b9bb60f15ae4e42f4efa15a)

File: blog/src/app/components/register/register.component.ts (hunk 2 above)
Comment (ShugKnight24) [reply blog/src/app/components/register/register.component.ts:15]: resolved by [b123d74](https://github.com/ShugKnight24/angular_nest_blog/pull/8/commits/b123d74634ca4aa8f268a17cbd1fe21f76c2c07a)

**Comment by ShugKnight24** [comment]
Dependent on #11, merge that in first

**Pull request merged**